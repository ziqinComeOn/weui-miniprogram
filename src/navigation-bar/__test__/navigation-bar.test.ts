import path from 'path'
import simulate from 'miniprogram-simulate'

describe('navigation-bar', () => {
    const navigationBar = simulate.load(
        path.resolve(__dirname, '../navigation-bar'),
        'mp-navigation-bar'
    )

    test('basic', async () => {
        const comp = simulate.render(
            simulate.load({
                usingComponents: {
                    'mp-navigation-bar': navigationBar
                },
                template: `
                    <mp-navigation-bar loading="{{true}}" title="UI组件库" back="{{true}}"></mp-navigation-bar>
                `
            })
        )
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
    })
})
