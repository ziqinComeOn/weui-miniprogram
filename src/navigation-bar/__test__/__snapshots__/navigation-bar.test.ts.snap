// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation-bar basic 1`] = `
<main>
  <mp-navigation-bar>
    <wx-view
      class="weui-navigation-bar "
    >
      <wx-view
        class="weui-navigation-bar__placeholder ios"
        style="padding-top: 20px;visibility: hidden;"
      />
      <wx-view
        class="weui-navigation-bar__inner ios"
        style="padding-top: 20px; color: ;background: ;;;;"
      >
        <wx-view
          class="weui-navigation-bar__left"
          style=""
        >
          <wx-view
            class="weui-navigation-bar__buttons"
          >
            <wx-view
              class="weui-navigation-bar__button weui-navigation-bar__btn_goback"
              bind:tap="back"
            />
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-navigation-bar__center"
        >
          <wx-view
            class="weui-navigation-bar__loading"
          >
            <wx-view
              class="weui-loading"
              style="width:rpx;height:rpx;"
            />
          </wx-view>
          <wx-text>
            UI组件库
          </wx-text>
        </wx-view>
        <wx-view
          class="weui-navigation-bar__right"
        />
      </wx-view>
    </wx-view>
  </mp-navigation-bar>
</main>
`;
